### ISSUE TITLE
Support for unix domain socket?

### ISSUE THREAD

**Issue by dvtan** [RobertoPrevato/BlackSheep#13]
Will there be support for unix domain socket?

Example use case: nginx is in front of the app server, all running on the same machine.

**Comment by RobertoPrevato** [comment]
Yes, I am planning the following:

* I will separate the HTTP server component from the web framework
- * the web framework will still be called "BlackSheep"
- * the http server will 

While doing this, I will make BlackSheep ASGI compatible, so it will be possible to run it on top of `uvicorn`. This will add support for unix domain sockets, and achieve separation of concerns between framework and server parts.
This is a similar path that Philip Jones followed for Quart and Hypercorn.

In the meanwhile, I will also update my own http server implementation to support unix domain sockets.

I recently explained the reasons why I didn't make BlackSheep ASGI compatible from the beginning, here: [https://robertoprevato.github.io/Presenting-BlackSheep/](https://robertoprevato.github.io/Presenting-BlackSheep/).

**Comment by RobertoPrevato** [comment]
Hi @dvtan, now BlackSheep adheres to ASGI. I removed the previous HTTP server implementation, so now it focuses on being a good web framework, and http client. Indirectly, **it supports Unix socket**, depending on the ASGI server used with it.

For now I tested it only with #uvicorn, it should work with #daphne and #hypercorn, too.

Compared to alternatives, You might find interesting these unique features in BlackSheep:
* Support for dependency injection
* Model binding inspired by ASP.NET Core
* Strategy to implement authentication and authorization
